src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java

```diff
@@ -195,25 +206,171 @@ private static DriverManagerType setupBrowserDriver(TestExecutionContext testExe
+                            logOutput.append(frame.toString());
+                        }
+                    }).awaitCompletion();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        Matcher matcher = versionPattern.matcher(logOutput.toString());
+        if (matcher.find()) {
+            return matcher.group(1);
+        }
+        return null;
+    }
+
+    private static String getLocalBrowserVersionFor(String browserType) {
+        String binaryPath = "";
+        switch (browserType) {
+            case "chrome":
+                binaryPath = getChromeBinaryPath();
+                break;
+            case "firefox":
+                binaryPath = getFirefoxBinaryPath();
+                break;
+            case "safari":
+                return getSafariVersion();
+            default:
+                throw new InvalidTestDataException("Invalid browser : " + browserType);
+        }
+        ChromeOptions chromeOptions = new ChromeOptions();
+        chromeOptions.setBinary(binaryPath);
+        ProcessBuilder processBuilder = new ProcessBuilder();
+        processBuilder.command(binaryPath, "--version");
+        String version = "";
+        try {
+            Process process = processBuilder.start();
+            BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            String line = reader.readLine();
+            String[] split = line.split(" ");
+            version = split[split.length - 1];
+            LOGGER.info("Browser Version in system : " + version);
+            process.destroy();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return version;
+    }
+
+    private static String getChromeBinaryPath() {
+        String os = System.getProperty("os.name").toLowerCase();
+        switch (os) {
+            case "win":
+                return System.getenv("ProgramFiles") + "\\Google\\Chrome\\Application\\chrome.exe";
+            case "mac os x":
+                return "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome";
+            case "nix":
+            case "nux":
+            case "linux":
+                return "/usr/bin/google-chrome";
+            default:
+                throw new IllegalStateException("Unsupported operating system: " + os);
+        }
+    }
+
+    private static String getFirefoxBinaryPath() {
+        String os = System.getProperty("os.name").toLowerCase();
+        switch (os) {
+            case "win":
+                return System.getenv("ProgramFiles") + "\\Firefox\\Application\\firefox.exe";
+            case "mac":
+                return "/Applications/Firefox.app/Contents/MacOS/firefox";
+            case "nix":
+            case "nux":
+            case "linux":
+                return "/usr/bin/firefox";
+            default:
+                throw new IllegalStateException("Unsupported operating system: " + os);
```

Comment:
like Unsupported operating system for firefix in +os

Reply:
updated